lib/Gateway.js

```diff
@@ -786,6 +807,77 @@ function addRgbDimmer (cfg, node, valueId) {
   }
 
   return false
+} */
+
+function addRgbColorSwitch (cfg, node, valueId) {
+  cfg = copy(cfg)
+  cfg.discovery_payload.rgb_state_topic = this.mqtt.getTopic(
+    this.valueTopic(node, valueId)
+  )
+  cfg.discovery_payload.rgb_command_topic =
+    cfg.discovery_payload.rgb_state_topic + '/set'
+
+  // The following part of code, checks if ML or Binary works. If one exists the other
+  let brightnessValue = false
+  let switchValue = false
+  if (node.values.hasOwnProperty(`38-${valueId.endpoint}-currentValue`)) {
```

Comment:
don't use hasOwnProperty here

Reply:
after slack chat Thanks for help and explanation!